en/ros/external_position_estimation.md

```diff
@@ -10,9 +10,20 @@ The system can then be used for applications such as position hold indoors or wa
 
 For vision, the MAVLink message used to send the pose data is [VISION_POSITION_ESTIMATE](https://mavlink.io/en/messages/common.html#VISION_POSITION_ESTIMATE) and the message for all motion capture systems is [ATT_POS_MOCAP](https://mavlink.io/en/messages/common.html#ATT_POS_MOCAP) messages.
 
-The mavros ROS-MAVLink interface has default implementations to send these messages. They can also be sent using pure C/C++ code and direct use of the MAVLink library. The ROS topics are: `mocap_pose_estimate` for mocap systems and `vision_pose_estimate` for vision. Check [mavros_extras](http://wiki.ros.org/mavros_extras) for further info.
+The mavros ROS-MAVLink interface has default implementations to send these messages. They can also be sent using pure C/C++ code and direct use of the MAVLink library. The ROS topics are: `/mavros/mocap/pose` for mocap systems and `/mavros/vision_pose/pose` for vision. Check [mavros_extras](http://wiki.ros.org/mavros_extras) for further info.
+
+## EKF2 Tuning for External Position Estimate
+
+A set of paramters need to be set in order for EKF2 to use the external position estimation. You can use QGroundContorl (QGC) to easilt adjust theses paramters. All of the following paramters can be found in the EKF2 tab in QGC.
+
+* External position estimate can be enabled by setting the `EKF_AID_MASK` to enable position and yaw fusion
+* To use the external height estimate for altitude estimation, set `EKF2_HGT_MODE` to use vision
+* Adjust the `EKF2_EV_DELAY` paramter according to how fast you receive the external position data with respect to the flight controller's IMU. Reduce the value of this paramter if you are getting external data at high rate, e.g. `50ms`
```

Comment:
This is not entirely correct. This value actually represents how far off the timestamp of the measurement is off from the "actual" time it was captured at. It can technically be set to 0 if there is correct timestamping (not just arrival time) and timesync (e.g NTP) between mocap and ROS computers. In reality, this needs some empirical tuning since delays in the entire Mocap->PX4 chain are very setup specific and there is rarely a well setup system with an entirely synchronised chain.

Empirical tuning involves looking at the EKF innovations during dynamic maneuvers, and doing a parameter search for this value which yields the lowest innovations.

Reply:
Probably I interpreted this incorrectly, but according to the docs [here](http://dev.px4.io/en/advanced/parameter_reference.html), `EKF2_EV_DELAY` is the Vision Position Estimator delay relative to IMU measurements.

Reply:
@mhkabir is right. It is the delay and has nothing to do with the rate.
You could also check for that delay in the logs by looking at the IMU rates and the EV rates check the offset.

Reply:
@ChristophTobler What I understood from @mhkabir is that it is not related to IMU!! Instead it is related to the correct stamping of the mocap data. How is it related to the IMU then?

I guess it is related to the fusion using the EKF, but now I'm not sure whether the `EKF2_EV_delay` is the time offset between the IMU and vision stamping, or something else!

Reply:
It is related to the IMU as the IMU time stamping is the "base clock" for the EKF.

>I guess it is related to the fusion using the EKF, but now I'm not sure whether the `EKF2_EV_delay` is the time offset between the IMU and vision stamping, or something else!

It's exactly that.

Reply:
It's the difference between the "base clock" as Christoph describes.

Reply:
You can get a rough estimate by looking at a look (see picture)
![ev_delay](https://user-images.githubusercontent.com/14769636/44080656-fc2ffcf0-9fac-11e8-8303-38a776969b53.png)

Reply:
@mzahana So can you fix up according to @mhkabir comment above : https://github.com/PX4/Devguide/pull/573#discussion_r209512837

Reply:
@ChristophTobler Perhaps a dumb question, but how do you get this log graph? Specifically, the legend at the bottom for the red and green lines has the same text, so looks like you're comparing like with like (even though I know one must be the IMU rate and the other must be the external vision rate)